File: Study/Programming Skills/01. Merge Strings Alternatively/main.go
package main

import "fmt"

func main() {
  fmt.Println(mergeAlternately("abc", "pqr"))
  fmt.Println(mergeAlternately("ab", "pqrs"))
  fmt.Println(mergeAlternately("abcd", "pq"))
}

func mergeAlternately(word1 string, word2 string) string {
  var result string
  for i := 0; i < len(word1) || i < len(word2); i++ {
    if i < len(word1) {
      result += string(word1[i])
    }
    if i < len(word2) {
      result += string(word2[i])
    }
  }
  return result
}

// Test Case
// mergeAlternately("abc", "pqr") => "apbqcr"
// mergeAlternately("ab", "pqrs") => "apbqrs"
// mergeAlternately("abcd", "pq") => "apbqcd"

File: Study/Programming Skills/01. Merge Strings Alternatively/main.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function (word1, word2) {
  let result = '';
  for (let i = 0; i < Math.max(word1.length, word2.length); i++) {
    if (word1[i]) result += word1[i];
    if (word2[i]) result += word2[i];
  }
  return result;
};


// Test cases
console.log(mergeAlternately('abc', 'pqr')); // "apbqcr"
console.log(mergeAlternately('ab', 'pqrs')); // "apbqrs"
console.log(mergeAlternately('abcd', 'pq')); // "apbqcd"
